libbpf-cargo/src/gen/btf.rs

```diff
@@ -805,30 +807,22 @@ impl<'s> GenBtf<'s> {
             }
         }
 
-        writeln!(
-            def,
-            r#"#[derive(Debug, Copy, Clone, Default, PartialEq, Eq)]"#
-        )?;
-        writeln!(def, r#"#[repr({signed}{repr_size})]"#)?;
-        writeln!(
-            def,
-            r#"pub enum {name} {{"#,
-            name = self.anon_types.type_name_or_anon(&t),
-        )?;
+        writeln!(def, r#"#[derive(Debug, Copy, Clone, Default)]"#)?;
+        writeln!(def, r#"pub struct {enum_name}({signed}{repr_size});"#,)?;
 
-        for (i, value) in t.iter().enumerate() {
-            if i == 0 {
-                writeln!(def, r#"    #[default]"#)?;
-            }
+        writeln!(def, r#"impl {enum_name} {{"#,)?;
+
+        for value in t.iter() {
+            writeln!(def, "    #[allow(non_upper_case_globals)]")?;
```

Comment:
Consider moving this hint to the overall type level instead of emitting it for each variant/constant.